File: DSA/Stack/Code/PrefixToPostfix.cs
// Prefix to Postfix in C#

using System;
using System.Collections.Generic;

class PrefixToPostfix {
    static bool IsOperator(char c) {
        return (c == '+' || c == '-' || c == '*' || c == '/');
    }
    
    static string PrefixToPostfixConversion(string prefix) {
        Stack<string> stack = new Stack<string>();
        int len = prefix.Length;
        
        for (int i = len - 1; i >= 0; i--) {
            char c = prefix[i];
            
            if (IsOperator(c)) {
                string op1 = stack.Pop();
                string op2 = stack.Pop();
                
                string temp = op1 + op2 + c;
                stack.Push(temp);
            } else {
                stack.Push(c.ToString());
            }
        }
        
        return stack.Pop();
    }
    
    static void Main() {
        string prefix = "-+ab*cd";
        
        Console.WriteLine("=== Prefix to Postfix ===\n");
        Console.WriteLine("Prefix Expression: " + prefix);
        
        string postfix = PrefixToPostfixConversion(prefix);
        
        Console.WriteLine("Postfix Expression: " + postfix + "\n");
        Console.WriteLine("Complexity: O(n)");
    }
}

File: DSA/Stack/Code/PrefixToPostfix.py
# Prefix to Postfix in Python

def is_operator(c):
    return c in ['+', '-', '*', '/']

def prefix_to_postfix(prefix):
    stack = []
    length = len(prefix)
    
    for i in range(length - 1, -1, -1):
        c = prefix[i]
        
        if is_operator(c):
            op1 = stack.pop()
            op2 = stack.pop()
            
            temp = op1 + op2 + c
            stack.append(temp)
        else:
            stack.append(c)
    
    return stack[0]

# Test
prefix = "-+ab*cd"

print("=== Prefix to Postfix ===\n")
print(f"Prefix Expression: {prefix}")

postfix = prefix_to_postfix(prefix)

print(f"Postfix Expression: {postfix}\n")
print("Complexity: O(n)")
